guidance/include/guidance/guidance_state_machine.hpp

```diff
@@ -71,15 +71,57 @@ namespace guidance
              */
             uint8_t getCurrentState();
 
+            virtual void StartUpState(Signal signal) = 0;
+            virtual void DriversReadyState(Signal signal) = 0;
+            virtual void ActiveState(Signal signal) = 0;
+            virtual void EngagedState(Signal signal) = 0;
+            virtual void InactiveState(Signal signal) = 0;
+            virtual void OffState(Signal signal) = 0;
+            // a local variable keeps the current state machine state
+            State current_guidance_state;
+
         private:
             /*!
              * \brief Actual state machine logic driven by signal enum
              */
             void onGuidanceSignal(Signal signal);
 
-            // a local variable keeps the current state machine state
-            State currentGuidanceState;
+    };
+
+    class Cadilac : public GuidanceStateMachine {
+        public:
+            void StartUpState(Signal signal);
+            void DriversReadyState(Signal signal);
+            void ActiveState(Signal signal);
+            void EngagedState(Signal signal);
+            void InactiveState(Signal signal);
+            void OffState(Signal signal);
+    };
 
+    class Lexus : public GuidanceStateMachine {
+        public:
+            void StartUpState(Signal signal);
+            void DriversReadyState(Signal signal);
+            void ActiveState(Signal signal);
+            void EngagedState(Signal signal);
+            void InactiveState(Signal signal);
+            void OffState(Signal signal);
+    };
+
+    class Factory {
+        public:
+            virtual GuidanceStateMachine* createCadilacInstance() = 0;
+            virtual GuidanceStateMachine* createLexusInstance() = 0;
+    };
+
+    class GuidanceStateMachineFactory : public Factory {
```

Comment:
Please keep the implementation of classes. In cpp files since that is the standard we have been using. It also is probably worth seperating the Factory classes into a seperate header file so that we have guidance_state_machine.h and guidance_state_machine_factory.h